File: Week2/node/flows/sortFlow.js
const { bubbleSort, insertionSort, mergeSort } = require("../algorithms/sort");
const { ask } = require("../utils/input");
const { printSortMenu } = require("../menu/menuPrinter");

async function runSortFlow(arr) {
  printSortMenu();
  const choice = await ask("Select sort type (1-3) or 0 to go back: ");

  if (choice === "0") return { next: "back" };

  if (choice === "1") {
    console.log("\nOriginal:", arr);
    console.log("Insertion Sort:", insertionSort(arr));
    return { next: "done" };
  }

  if (choice === "2") {
    console.log("\nOriginal:", arr);
    console.log("Bubble Sort:", bubbleSort(arr));
    return { next: "done" };
  }

  if (choice === "3") {
    console.log("\nOriginal:", arr);
    console.log("Merge Sort:", mergeSort(arr));
    return { next: "done" };
  }

  console.log("Invalid sort option.");
  return { next: "done" };
}

module.exports = { runSortFlow };

File: Week2/python/flows/sort_flow.py
from algorithms.sort import bubble_sort, insertion_sort, merge_sort
from menu.menu_printer import print_sort_menu


def run_sort_flow(arr):
    print_sort_menu()
    choice = input("Select sort type (1-3) or 0 to go back: ").strip()

    if choice == "0":
        return

    if choice == "1":
        print("\nOriginal:", arr)
        print("Insertion Sort:", insertion_sort(arr))
        return

    if choice == "2":
        print("\nOriginal:", arr)
        print("Bubble Sort:", bubble_sort(arr))
        return

    if choice == "3":
        print("\nOriginal:", arr)
        print("Merge Sort:", merge_sort(arr))
        return

    print("Invalid sort option.")
